internal/core/application/trade_service.go

```diff
@@ -326,13 +326,14 @@ func (t *tradeService) TradePropose(
 				return trade, nil
 			}
 
-			if !isValidTradePrice(swapRequest, tradeType, previewAmount) {
+			if !isValidTradePrice(swapRequest, tradeType, mkt) {
 				trade.Fail(
 					swapRequest.GetId(),
 					domain.ProposalRejectedStatus,
 					pkgswap.ErrCodeInvalidSwapRequest,
 					"bad pricing",
 				)
+				swapFail = trade.SwapFailMessage()
 				return trade, nil
```

Comment:
We should store the fee in the collected trade domain isn it_

Reply:
yes, but it's not in the scope of this PR. Don't want to add other changes to fix the ReportMarketFee rpc.